bindings/java/src/main/java/org/apache/opendal/Metadata.java

```diff
@@ -19,26 +19,98 @@
 
 package org.apache.opendal;
 
+import java.util.Date;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
 /**
  * Metadata carries all metadata associated with a path.
  */
-public class Metadata extends NativeObject {
-    protected Metadata(long nativeHandle) {
-        super(nativeHandle);
+@ToString
+@EqualsAndHashCode
```

Comment:
May we use `@Data` here to preserve getters. Somewhere only methods are allowed, and getters API is exposed before. When we can keep the API compatibility, we don't proactively break it.

Still, all the field can be `public final`, this is fine.